configure.ac

```diff
@@ -391,10 +391,10 @@ AS_IF([test "x$squid_opt_aufs_threads" != "x"],[
 
 ## TODO check if this is necessary, LT_INIT does these checks
 SQUID_AUTO_LIB(dl,[dynamic linking],[LIBDL])
-AS_IF([test "x$with_dl" != "xno"],[
-  CXXFLAGS="$LIBDL_CFLAGS $CXXFLAGS"
+SQUID_CHECK_LIB_WORKS(dl,[
   LDFLAGS="$LIBDL_PATH $LDFLAGS"
-  AC_CHECK_LIB(dl, dlopen)
+  AC_CHECK_LIB(dl, dlopen,[LIBDL_LIBS="-ldl"])
+  LIBS="$LIBDL_LIBS $LIBS"
```

Comment:
Do we need this `LIBS` update? AFAICT, the official code did not do that, some other SQUID_CHECK_LIB_WORKS() callers do not do that, and the PR description does not mention any libdl-related bug fixes. If we do need to update `LIBS` here, it may be a good idea to add a source code comment to explain why.

Reply:
Yes, `AC_CHECK_LIB` sets `LIBS` by default **unless** some other action is specified as the macro argument 2. For now this line exists to preserve the old behaviour in all followup build actions. Reviewing its removal is left to a followup.